src/Microsoft.AspNet.Authentication.Cookies/CookieAuthenticationHandler.cs

```diff
@@ -101,43 +101,28 @@ private Task<AuthenticationTicket> EnsureCookieTicket()
             return ticket;
         }
 
-        protected override async Task<AuthenticationTicket> HandleAuthenticateAsync()
+        protected override async Task<AuthenticateResult> HandleAuthenticateAsync()
         {
-            AuthenticationTicket ticket = null;
-            try
+            var ticket = await EnsureCookieTicket();
+            if (ticket == null)
             {
-                ticket = await EnsureCookieTicket();
-                if (ticket == null)
-                {
-                    return null;
-                }
-
-                var context = new CookieValidatePrincipalContext(Context, ticket, Options);
-                await Options.Events.ValidatePrincipal(context);
-
-                if (context.Principal == null)
-                {
-                    return null;
-                }
+                return AuthenticateResult.Failed("No ticket.");
```

Comment:
Yeah next iteration when I go through all of the cases, we can tweak all of the errors there